docs/introduction.org

```diff
@@ -0,0 +1,106 @@
+* Introduction
+
+Nowadays, multiple languages have emerged to compete with the well-established C language.
+In fact, C is widely used in many places, learned by a lot of engineers, and used for many purposes.
+C is a powerful tool, /but with great power comes great responsibility/, in fact, memory leaks can easily occur by mistake.
+
+That's why new languages come to find a better way to handle errors and memory.
+The Rust language relies strongly on its borrow checker to avoid safe memory but it comes at a cost: increased code complexity
+
+In another case, the Zig language has a purpose to replace the C language.
+Zig wants to simplify the programming development experience with no hidden control flow and no macro to avoid complex mistakes.
+In comparison with the Rust language, Zig uses explicit memory allocation to prevent memory leaks.
+
+In the sections below, certain aspects of this programming language will be explored in greater depth.
+
+** What is not this documentation
+This is *not* a beginner guide for Zig, this documentation wants to go in-depth in certain parts of the language.
+
+To understand what is described in this documentation, you should first read up on the basics of the Zig language, in particular, its guide, which contains a wealth of information.
+- [[https://zig.guide][Zig guide]]
+- [[https://ziglang.org/documentation/0.11.0][Zig documentation]]
+
+** How to use this documentation
+
+This documentation is built to guide to user through different examples.
+Those examples have been created to use literate programming, in fact, the documentation is built on Emacs with the org mode.
+You can follow the examples and execute it directly in Emacs.
+
+For that, you need to:
+1. Install the configuration created for that (see the section below)
+2. When all is configured, in Emacs you can select the example that you want to execute and use the command ~C-c C-c~.
+   The output will be created or updated below the source block. If you want more documentation, you can check the [[https://orgmode.org/worg/org-contrib/babel/intro.html][Org babel documentation]]
+
+You can follow the examples with the Zig version 0.12.
+
+** Build Emacs configuration
+
+*** Install Zig emacs org babel plugin
+Use ~C-c C-c~ (~ctrl-c ctrl-c~) to evaluate the code below.
+It will add a shell interpreter inside org-babel configuration, it will be used when shell commands are evaluated.
+
+#+begin_src emacs-lisp
+  (add-to-list 'org-babel-load-languages '(shell . t))
+  (org-babel-do-load-languages 'org-babel-load-languages org-babel-load-languages)
+#+end_src
+
+The shell command will clone an org babel zig plugin necessary to use this documentation with iterative programming.
```

Comment:
im not sure I understand this sentence, is iterative programming really what you wanted to say ?

Reply:
1. "to use execute" is weird.

2. I still dont get it, what is an org babel zig plugin ?

Reply:
Ah it's the ob-zig emacs li

Reply:
ok ty :) might be cool to maybe precise it or put a link to the gh or something like that